code/__HELPERS/bandetect.dm

```diff
@@ -0,0 +1,45 @@
+#define YOUNG 4
+
+
+/client/proc/join_date_check(var/y,var/m,var/d)
+	if (!current_year)
+		current_year = text2num(time2text(world.realtime, "YYYY"))
+	if (!current_month)
+		current_month = text2num(time2text(world.realtime, "MM"))
+	if (!current_day)
+		current_day = text2num(time2text(world.realtime, "DD"))
+	var/warn = 0
+	if (current_month == 1 && current_day <= YOUNG)
+		if (y == current_year - 1 && m == 12 && d >= 31 - (YOUNG - current_day))
```

Comment:
Get me to explain it on IRC; several calls are already made to the DB every time a player connects, another really simple one won't matter much.

orange's idea would work too, and possibly better than SQL, but you'd have to find or create a program for that.